components/segmenter/src/complex/lstm/matrix.rs

```diff
@@ -8,20 +8,24 @@ use core::ops::Range;
 use zerovec::ule::AsULE;
 use zerovec::ZeroSlice;
 
-// Polyfill float operations with libm in case we're no_std.
-#[allow(unused_imports)]
-use num_traits::Float;
-
-/// `tanh` computes the tanh function for a scalar value.
-#[inline]
-fn tanh(x: f32) -> f32 {
-    x.tanh()
+// This will be used in #[no_std] as f32::exp/f32::tanh are not in core.
+trait CoreFloat {
+    fn exp(self) -> Self;
+    fn tanh(self) -> Self;
 }
 
-/// `sigmoid` computes the sigmoid function for a scalar value.
-#[inline]
-fn sigmoid(x: f32) -> f32 {
-    1.0 / (1.0 + (-x).exp())
+// TODO: Find a way to do this without libm. There's an intrinsic
+// that produces `llvm.exp.f32`, but that's not exposed through
+// f32. Another option would be to declare `extern "Rust"` and
+// let clients provide these functions.
+impl CoreFloat for f32 {
+    fn exp(self) -> Self {
+        libm::expf(self)
```

Comment:
Question: before we introduced libm via num-traits, we had been using the f32::exp function. Now we're unconditionally using the libm version. Explain?

Reply:
Are we? In std mode this would fall back to `f32.exp()`, no?

Reply:
Before we were either using the inherent `f32::exp`, or `num_traits::Float::f32`, which is backed by `libm`. This is doing the exact same thing.

Reply:
Before we introduced `num_traits` we weren't `no_std`. With `std`, function resolution will use `f32::exp`, without it will use `CoreFloat::exp`.